### ISSUE TITLE
Add L1 pre-firing weights

### ISSUE THREAD

**Issue by bortigno** [UFLX2MuMu/Ntupliser#50]
Statring from CMSSW_9_4_13 we have L1 pre-firing weights for MC computed by a producer - for more details see:
https://github.com/cms-sw/cmssw/pull/25380

Some more information about the recipe is here:
https://twiki.cern.ch/twiki/bin/viewauth/CMS/L1ECALPrefiringWeightRecipe

**Comment by bortigno** [comment]
Start working on issue-50-l1-prefiring branch

**Comment by bortigno** [comment]
l1-prefiring weight included in mc. ops test ok. 
todo:
- [ ]  need to check if weights makes sense.
- [ ]  update test_crab_config and the main crab config

**Comment by bortigno** [comment]
This is the distribution of the l1prefiring weights for a ggH sample.
![Screenshot 2019-03-25 at 12 14 23](https://user-images.githubusercontent.com/8289819/54915605-d3465000-4ef7-11e9-8048-af3a5645a648.png)

**Comment by bortigno** [comment]
Opened a PR #56

**Comment by bortigno** [comment]
PR #56 merged. Closing the issue.

**Comment by bortigno** [comment]
Need to do the same for 2016. Working on branch `issue-50-l1-prefiring-2016`.

**Comment by bortigno** [comment]
Once I follow the wiki https://twiki.cern.ch/twiki/bin/viewauth/CMS/L1ECALPrefiringWeightRecipe and change in test/test_ntupliser_mc.py  test/test_ntupliser_data.py and crab/template/EDAnalyzer.py in the `process.prefiringweight` the DataEra to
```
DataEra = cms.string("2016BtoH")
```
the test crashes in the contracture. I suspect it is not related to the prefiring, but just to the test/test_ntupliser_mc.py. This config does not work with 2016. While UFDiMuonAnalyzer_2018_02_07.py does. I am checking for meaningful differences between the two. The main 2017 production has been done using a config very similar to test/test_ntupliser_mc.py

First I think I should remove `runMetCorAndUncFromMiniAOD` because it is not needed in 2016. I am not adding the JEC update because the ReReco should have the most updated ones.

If I change the cff from python/Analyzer_MC_cff to python/Analyzer_2017_94X_MC_cff it runs fine, so the problem is in the cff. Comparing them now. The only difference I see is in the muon SF that in the 2017 are son file and in 2016 root files. I remembered I tried to keep the use of root possible when I moved to 2017 json in
